suplemon/viewer.py

```diff
@@ -50,76 +51,38 @@ def __init__(self, app, window):
         self.x_scroll = 0
         self.cursors = [Cursor()]
 
-        # Lexer for translating tokens to strings
-        self.lexer = None
-        # Built in syntax definition (for commenting etc.)
-        self.syntax = None
-        # Normal Pygments lexer
-        self.pygments_syntax = None
+        # Copy/paste buffer
+        self.buffer = []
+
+        # Runnable methods
+        self.operations = {
+            "arrow_right": self.arrow_right,              # Arrow Right
+            "arrow_left": self.arrow_left,                # Arrow Left
+            "arrow_up": self.arrow_up,                    # Arrow Up
+            "arrow_down": self.arrow_down,                # Arrow Down
+            "jump_left": self.jump_left,                  # Ctrl + Left
+            "jump_right": self.jump_right,                # Ctrl + Right
+            "jump_up": self.jump_up,                      # Ctrl + Up
+            "jump_down": self.jump_down,                  # Ctrl + Down
+            "page_up": self.page_up,                      # Page Up
+            "page_down": self.page_down,                  # Page Down
+            "home": self.home,                            # Home
+            "end": self.end,                              # End
+            "find": self.find_query                       # Ctrl + F
+        }
```

Comment:
This does seem more robust so I'll transition to the Operations class when I have time. Thanks for the good examples to get started with! One question though, how does `run_operation` know which method to call?